runtime/src/main/java/org/corfudb/runtime/view/LayoutManagementView.java

```diff
@@ -99,17 +100,26 @@ public void handleFailure(IReconfigurationHandlerPolicy failureHandlerPolicy,
      * This bootstraps the Layout server with the existing layout.
      *
      * @param endpoint New node endpoint.
-     * @return Completable Future which completes when the node's layout is bootstrapped.
+     * @return Completable Future which completes with true if the node's layout is bootstrapped,
+     * false if it has already been bootstrapped. The future completes exceptionally if a time out
+     * has occurred.
      */
     public CompletableFuture<Boolean> bootstrapNewNode(String endpoint) {
 
         // Bootstrap the to-be added node with the old layout.
         Layout layout = new Layout(runtime.getLayoutView().getLayout());
-        return runtime.getLayoutView().bootstrapLayoutServer(endpoint, layout)
-                .thenApply(result -> {
-                    log.info("bootstrapNewNode: New node {} bootstrapped.", endpoint);
-                    return true;
-                });
+
+        CompletableFuture<Boolean> future = new CompletableFuture<>();
+
+        try {
+            boolean bootstrapped = runtime.getLayoutView().bootstrapLayoutServer(endpoint, layout);
+            future.complete(bootstrapped);
+            return future;
+        }
+        catch (TimeoutException toe){
+            future.completeExceptionally(toe);
+            return future;
+        }
```

Comment:
This seems hacky, and besides why this is needed?  `bootstrapLayoutServer` has an exceptionally block:

```java
.exceptionally(throwable -> {
        try {
                CFUtils.unwrap(throwable, AlreadyBootstrappedException.class);
        } catch (AlreadyBootstrappedException e) {
                log.info("bootstrapLayoutServer: Layout Server {} already bootstrapped.", endpoint);
        }
        return true;
})
```
Notice this `CFUtils.unwrap(throwable, AlreadyBootstrappedException.class);` It will throw a `RutimeException` in case of exceptions other than `AlreadyBootstrappedException`, which makes the returned future exceptionally completed . Thus in this `bootstrapNewNode` method you will get a exceptionally completed future of `RutimeException` with a `TimeoutException` wrapped inside.